manifests/repo/el.pp

```diff
@@ -6,14 +6,14 @@
   if $jenkins::lts {
     $baseurl = "${jenkins::repo::base_url}/redhat-stable/"
   } else {
-    $baseurl = "${jenkins::repo::base_url}/redhat/"
+    $baseurl = "${jenkins::repo::base_url}/redhat/${jenkins::repo::gpg_key_filename}"
   }
 
   yumrepo {'jenkins':
     descr    => 'Jenkins',
     baseurl  => $baseurl,
     gpgcheck => 1,
-    gpgkey   => "${baseurl}${jenkins::repo::gpg_key_filename}",
+    gpgkey   => "${baseurl}/redhat/${jenkins::repo::gpg_key_filename}",
```

Comment:
This doesn't looks right either. You end up with `/redhat/redhat/jenkins.io.key`.